spec/controllers/nodes_controller_spec.rb

```diff
@@ -95,5 +95,19 @@
       end
       expect(response.status).to eq 302
     end
+
+    context "a role fails to be assigned" do
+      # rubocop:disable AnyInstance
+      before do
+        allow_any_instance_of(Minion).to receive(:assign_role).and_return(false)
```

Comment:
Note that this forces us to disable `AnyInstance` rubocop check, because the usage of `allow_any_instance_of` is discouraged. However at this time, we haven't still instantiated the elements. We could do so manually and provide them as an argument to the bootstrap action, but it looked strange.